vernac/library.ml

```diff
@@ -522,6 +534,11 @@ let save_library_struct ~output_native_objects dir =
     error_recursively_dependent_library dir;
   sd, md, vmlib, ast
 
+let save_library dir : library_t =
+  let sd, md, vmlib, _ast = save_library_struct ~output_native_objects:false dir in
+  let digest = Safe_typing.Dvo_or_vi (Digest.string "") in
+  mk_library sd md digest Univ.ContextSet.empty (Vmlibrary.inject vmlib)
```

Comment:
@ppedrot can you confirm this is correct w.r.t. the new `Vmlibrary` stuff?

Reply:
I think so. Unrelatedly the dummy digest part is slightly more concerning to me, we should have an option there or something to signal that the data didn't come from a vo file.

Reply:
(Also as-is this will never work with native, which is another reason why we should have a distinct library type.)

Reply:
Let me indeed fix the digest stuff before this is merged.

Actually native could be made work with this in the same way: have the kernel relay a .cmxs request to the document manager / whoever is coordinating the filesystem layout.

Indeed it seems to me that a few parts of Coq could benefit from exposing a caching layer (that's the way I see the .vo files, as a cache), for example in #17674 ; I wanted to discuss this in the call actually.